components/script/dom/globalscope.rs

```diff
@@ -2878,124 +2882,184 @@ impl GlobalScope {
+            line_number,
+            introduction_type,
+        ));
 
-        unsafe {
-            rooted!(in(*cx) let mut compiled_script = std::ptr::null_mut::<JSScript>());
-            match code {
-                SourceCode::Text(text_code) => {
-                    let mut options = CompileOptionsWrapper::new_raw(*cx, filename, line_number);
-                    if let Some(introduction_type) = introduction_type {
-                        options.set_introduction_type(introduction_type);
-                    }
+        let record = if compiled_script.get().is_null() {
+            rooted!(in(*cx) let mut exception = UndefinedValue());
 
-                    debug!("compiling dom string");
-                    compiled_script.set(Compile1(
-                        *cx,
-                        options.ptr,
-                        &mut transform_str_to_source_text(&text_code.str()),
-                    ));
+            assert!(unsafe { JS_GetPendingException(*cx, exception.handle_mut()) });
+            unsafe { JS_ClearPendingException(*cx) };
 
-                    if compiled_script.is_null() {
-                        debug!("error compiling Dom string");
-                        report_pending_exception(cx, true, InRealm::Entered(&ar), can_gc);
-                        return Err(JavaScriptEvaluationError::CompilationFailure);
-                    }
-                },
-                SourceCode::Compiled(pre_compiled_script) => {
-                    let options = InstantiateOptions {
-                        skipFilenameValidation: false,
-                        hideScriptFromDebugger: false,
-                        deferDebugMetadata: false,
-                        eagerDelazificationStrategy_: DelazificationOption::OnDemandOnly,
-                    };
-                    let script = InstantiateGlobalStencil(
-                        *cx,
-                        &options,
-                        *pre_compiled_script.source_code,
-                        ptr::null_mut(),
-                    );
-                    compiled_script.set(script);
-                },
-            };
+            Err(RethrowError::new(Heap::boxed(exception.get())))
+        } else {
+            Ok(NonNull::new(*compiled_script).expect("Can't be null"))
+        };
 
-            assert!(!compiled_script.is_null());
+        let mut script = ClassicScript {
+            record,
+            url,
+            fetch_options,
+            muted_errors,
+            source: source_code,
+            external,
+            unminified_dir: self.unminified_js_dir(),
+        };
+        unminify_js(&mut script);
+        script
+    }
 
-            rooted!(in(*cx) let mut script_private = UndefinedValue());
-            JS_GetScriptPrivate(*compiled_script, script_private.handle_mut());
+    /// <https://html.spec.whatwg.org/multipage/#run-a-classic-script>
+    #[expect(unsafe_code)]
+    pub(crate) fn run_a_classic_script(
```

Comment:
Please also add spec steps that are currently missing and add corresponding TODO's. See https://github.com/servo/servo/blob/60b9a03c816c8aea5056555a6d92402b96ea3cc6/components/script/dom/event.rs#L325-L329 as example